test/daemon_test.rb

```diff
@@ -0,0 +1,94 @@
+require_relative 'test_helper'
+require 'dynflow/rails/daemon'
+
+class DaemonTest
+  describe 'Memory watching daemon' do
+    include PlanAssertions
+    include Dynflow::Testing::Assertions
+    include Dynflow::Testing::Factories
+    include TestHelpers
+
+    before do
+      @dynflow_memory_watcher = Minitest::Mock.new('memory_watcher')
+      @daemons = Minitest::Mock.new('daemons')
+      @daemon = Dynflow::Rails::Daemon.new(
+        @dynflow_memory_watcher,
+        @daemons
+      )
+      @world_class = Minitest::Mock.new('dummy world factory')
+      @dummy_world = Dynflow::Testing::DummyWorld.new
+      @dummy_world.stub(:auto_execute) { nil }
```

Comment:
The tests will need some attention, please note Dynflow uses "plain" `minitest` instead of `mocha` which does stubbing a bit differently http://docs.seattlerb.org/minitest/Object.html#method-i-stub